### DISCUSSION TITLE
Excluding single property from mapping

### DISCUSSION THREAD

**Discussion by twallmey** [mapstruct/mapstruct#3550] [Q&A]
Hi,
I've got several mappers (with a global Mapper-Config) that map between different object types. Some of these object types do have a special property A that I would like to exclude from mapping globally. How to do this the best way - is there any option I haven't seen so far that enables me to do this via MapperConfig?

Exception I get:
Unmapped target property: "A". Mapping from property....

My central mapper config:
@Mapper(
    config = MapstructConfig.class,
    componentModel = MappingConstants.ComponentModel.SPRING
)

**Comment by filiphr** [comment]
Perhaps you can use something like [Shared configurations](https://mapstruct.org/documentation/stable/reference/html/#shared-configurations). However, if your structure is not like expected there is no way to achieve what you are looking for.

You might be able to achieve this using a custom `AccessorNamingStrategy` where you would ignore this special property on your types.

**Comment by twallmey** [comment]
Thanks for your response - even so it's disappointing to here...

I've added the following annotion to all my mappers to skip this special property (persistenceContext):

`@Mapper(
    config = MapstructConfig.class,
    componentModel = MappingConstants.ComponentModel.SPRING
)
public interface SkillLevelMapperPhysical extends VersionConvertable<SkillLevelV2, SkillLevel> {
  SkillLevelMapperPhysical INSTANCE = Mappers.getMapper(SkillLevelMapperPhysical.class);
  ;

  @Mapping(
      target = "persistenceContext",
      ignore = true
  )
  SkillLevel convert(SkillLevelV2 from, @Context CycleAvoidingMappingContext context);

  @InheritInverseConfiguration
  SkillLevelV2 inverseConvert(SkillLevel from, @Context CycleAvoidingMappingContext context);
}`

Nevertheless I still get a warning that says that this property is unmapped. Am I doing something wrong that the warning is still mentioned although the explicit "ignore"-flag is set?

> [WARNING] /C:/project/skb/skb/skb-bean-version/src/main/java/com/service/dkp/skb/knowledge/model/version/map/product/appliance/problem/SkillLevelMapperPhysical.java:[33,14] Unmapped target property: "persistenceContext". Mapping from Collection element "ProductGroupV2 symptoms[].faultcodes[].productGroups" to "ProductGroup symptoms[].faultcodes[].productGroups".

**Reply by filiphr** [reply]
The definition that you are providing is only for the `SkillLevel` and not for the nested objects. You'll have to define mappings on all levels to define this.

For this particular use case I would suggest to use a custom `AccessorNamingStrategy`.

**Reply by twallmey** [reply]
I've tried to use a MappingExclusionProvider but it seems that attributes of my beans (s. attached sceenshot) are not reported to its overriden method - only beans that are mapped are reported to this method:

` @Override
    public boolean isExcluded(TypeElement typeElement) {`

![image](https://github.com/mapstruct/mapstruct/assets/47970789/ce467be5-ac56-4063-841b-52f26aa7b048)


Hence, I could only exclude a complete bean type from mapping (e.g. "symptom") but not only some of their attributes. According to the screenshot I would like to exclude the attribute "persistenceContext" (1) only from mapping. But neither adding the annotation "@Mapping ignore" to each of my beans nor providing an exclusion filter (s. above) results in success.

Could you tell me what I am doing wrong and/or how I could exclude dedicated attributes/properties of my beans?

**Reply by filiphr** [reply]
You should look at the `AccessorNamingStrategy` not `MappingExclusionProvider`
